deltaspike/core/impl/src/main/java/org/apache/deltaspike/core/impl/config/TypedResolverImpl.java

```diff
@@ -504,6 +530,66 @@ else if (Double.class.equals(configEntryType))
         {
             result = Double.parseDouble(value);
         }
+        else if (Character.class.equals(configEntryType))
```

Comment:
think we must take care of the packages, we likely don't want awt package to be loaded there because there is a security manager or javaagent for example.
think it can be better to provide the classes and enable to provide a Map<Class<?>, Converter<?>> to the resolver.
by default - for implicit resolvers - we can try to load all the ones we support - even if it should be per package group likely to avoid to fail 4 times when once is enough?. This default loading should likely be done in the config extension or something like that and cached to be reused/reusable (if the extension is injected). Manual resolvers can be loaded without any additional converter (primitives/wrappers being the one we must support OOTB only IMHO).

Side note: for the story behind this way to do, we did the exact same thing in xbean with its converts == load them all by default, but in practise it is very bothering and letting them be configurable and loaded at demand only is way more relevant on user land in general.

Reply:
Only took me 2.5 years to come back to this. ^-^'

In general, this makes sense. I'm currently trying to set up DeltaSpike v2.0.0-SNAPSHOT for development, so I can clean this up and finally get it in a state it can be merged. I've encountered an issue that I wanted to ask about, though.

In a sandbox project, it looks like OpenWebBeans is trying to find and inject a `java.util.regex.Pattern`, rather than leaving it for the DeltaSpike Configuration module to handle. 🤔

I don't see any immediate issues with how things are set up, though. Sorry to ask, but are you aware of any breaking changes that may prevent this from working:

https://gist.github.com/SethFalco/82e16523963880c609b3a9b09ac6fd67

Reply:
I guess it is cause you didn't set a converter on the injection and there is no default built-in producer of such a type (this is the challenge with this PR, to modify org.apache.deltaspike.core.impl.config.ConfigurationExtension#collectDynamicTypes for default case when there is a known converter in the converter "registry" but avoid to load and define everything upfront which would make the whole container slower for a very poor gain for most apps)

Reply:
Ahh, right this makes a little more sense to me now.

> we likely don't want awt package to be loaded there because there is a security manager or javaagent for example

I've at least handled this part.

For now, the existing converters (primitives) are all supported out of the box, while the new proposed converters can be loaded on demand with a producer.

```java
@Produces
public static PatternConverter getPatternConverter() {
    return new PatternConverter();
}
```

I'm now able to use DeltaSpike + these converters as I'd expect when imported into another project.

> provide a Map<Class, Converter> to the resolver

Could I make you elaborate on this part? Unfortunately, I did try to look around the configuration extension, but wasn't sure how to apply this.